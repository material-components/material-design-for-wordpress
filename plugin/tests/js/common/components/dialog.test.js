/**
 * Copyright 2021 Google LLC
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *      http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

/**
 * External dependencies
 */
import '@testing-library/jest-dom/extend-expect';
import { render } from '@testing-library/react';

/**
 * Internal dependencies
 */
import Dialog from '../../../../assets/src/common/components/dialog';

/**
 * Render the component.
 *
 * @param {Object} props - Component props
 * @return {Function} A functional component.
 */
const setup = props => {
	return render( <Dialog { ...props } /> );
};

const baseProps = {
	open: true,
	title: '<h3>Dialog Title</h3>',
	content: '<p>Dialog Content</p>',
	actions: '<button>Save</button>',
};

describe( 'Common Components: Dialog', () => {
	it( 'should match snapshot', () => {
		const wrapper = setup( baseProps );
		expect( wrapper ).toMatchSnapshot();
	} );

	it( 'should match snapshot when dialog is not open', () => {
		const wrapper = setup( { ...baseProps, ...{ open: false } } );
		expect( wrapper ).toMatchSnapshot();
	} );
} );
