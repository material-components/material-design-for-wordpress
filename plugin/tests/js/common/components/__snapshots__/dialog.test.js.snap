// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Common Components: Dialog should match snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="mdc-dialog mdc-dialog--open"
      >
        <div
          class="mdc-dialog__container"
        >
          <div
            aria-describedby="dialog-content-0"
            aria-labelledby="dialog-title-0"
            aria-modal="true"
            class="mdc-dialog__surface"
            role="alertdialog"
          >
            <h2
              class="mdc-dialog__title"
              id="dialog-title-0"
            >
              &lt;h3&gt;Dialog Title&lt;/h3&gt;
            </h2>
            <div
              class="mdc-dialog__content"
              id="dialog-content-0"
            >
              &lt;p&gt;Dialog Content&lt;/p&gt;
            </div>
            <div
              class="mdc-dialog__actions"
            >
              &lt;button&gt;Save&lt;/button&gt;
            </div>
          </div>
        </div>
        <div
          class="mdc-dialog__scrim"
        />
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="mdc-dialog mdc-dialog--open"
    >
      <div
        class="mdc-dialog__container"
      >
        <div
          aria-describedby="dialog-content-0"
          aria-labelledby="dialog-title-0"
          aria-modal="true"
          class="mdc-dialog__surface"
          role="alertdialog"
        >
          <h2
            class="mdc-dialog__title"
            id="dialog-title-0"
          >
            &lt;h3&gt;Dialog Title&lt;/h3&gt;
          </h2>
          <div
            class="mdc-dialog__content"
            id="dialog-content-0"
          >
            &lt;p&gt;Dialog Content&lt;/p&gt;
          </div>
          <div
            class="mdc-dialog__actions"
          >
            &lt;button&gt;Save&lt;/button&gt;
          </div>
        </div>
      </div>
      <div
        class="mdc-dialog__scrim"
      />
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Common Components: Dialog should match snapshot when dialog is not open 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="mdc-dialog"
      >
        <div
          class="mdc-dialog__container"
        >
          <div
            aria-describedby="dialog-content-1"
            aria-labelledby="dialog-title-1"
            aria-modal="true"
            class="mdc-dialog__surface"
            role="alertdialog"
          >
            <h2
              class="mdc-dialog__title"
              id="dialog-title-1"
            >
              &lt;h3&gt;Dialog Title&lt;/h3&gt;
            </h2>
            <div
              class="mdc-dialog__content"
              id="dialog-content-1"
            >
              &lt;p&gt;Dialog Content&lt;/p&gt;
            </div>
            <div
              class="mdc-dialog__actions"
            >
              &lt;button&gt;Save&lt;/button&gt;
            </div>
          </div>
        </div>
        <div
          class="mdc-dialog__scrim"
        />
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="mdc-dialog"
    >
      <div
        class="mdc-dialog__container"
      >
        <div
          aria-describedby="dialog-content-1"
          aria-labelledby="dialog-title-1"
          aria-modal="true"
          class="mdc-dialog__surface"
          role="alertdialog"
        >
          <h2
            class="mdc-dialog__title"
            id="dialog-title-1"
          >
            &lt;h3&gt;Dialog Title&lt;/h3&gt;
          </h2>
          <div
            class="mdc-dialog__content"
            id="dialog-content-1"
          >
            &lt;p&gt;Dialog Content&lt;/p&gt;
          </div>
          <div
            class="mdc-dialog__actions"
          >
            &lt;button&gt;Save&lt;/button&gt;
          </div>
        </div>
      </div>
      <div
        class="mdc-dialog__scrim"
      />
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
